compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -88,8 +88,54 @@ pub const UNKNOWN_COLUMN_NUMBER: c_uint = 0;
 
 pub const NO_SCOPE_METADATA: Option<&DIScope> = None;
 
-#[derive(Copy, Debug, Hash, Eq, PartialEq, Clone)]
-pub struct UniqueTypeId(Symbol);
+mod unique_type_id {
+    use super::*;
+    use rustc_arena::DroplessArena;
+
+    #[derive(Copy, Hash, Eq, PartialEq, Clone)]
+    pub(super) struct UniqueTypeId(u32);
```

Comment:
This type used to be a wrapper around `Symbol` with a `Debug` derive that used the global interner, but the `Symbol` would actually be interned by a separate interner.